src/track/globaltrackcache.cpp

```diff
@@ -594,27 +596,32 @@ void GlobalTrackCache::resolve(
                 trackRef.getCanonicalLocation());
         if (strongPtr) {
             // Cache hit
-            if (debugLogEnabled()) {
-                kLogger.debug()
-                        << "Cache hit - found track by canonical location"
-                        << trackRef.getCanonicalLocation()
-                        << strongPtr.get();
-            }
-            auto cachedTrackRef = validateAndCanonicalizeRequestedTrackRef(
-                    trackRef,
-                    *strongPtr);
+
             // Multiple tracks may reference the same physical file on disk
-            if (!trackRef.hasId() || trackRef.getId() == cachedTrackRef.getId()) {
+            auto cachedTrackRef = createTrackRef(*strongPtr);
+            if (trackRef.getLocation() == cachedTrackRef.getLocation()) {
+                if (debugLogEnabled()) {
```

Comment:
This condition feels redundant - it seems to check whether debug logs are enable to then send logs to debug anyway. AFAIK, there is no value in doing so ass debug logging would translate to no-op anyway, so this looks like unnecessary optimisation.

Note that `debugLogEnabled` seems to also check for `kLogEnabled` but this is not used anywhere, meaning that it duplicate what you would already get with `log-level=debug` anway, without needed recompilation

Reply:
This debug message is disable by default. In other place this message is just left as a comment. In this class the original author has coded around this. I have no intererst to change it here. 

When debugging. I actually prefere to enable/disable debug messages via //. I have normally no overview what I will see if after I bulk enable debugging with such a feature.